runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -73,18 +73,25 @@ public Object getPayload(CorfuRuntime runtime) {
                     if (data == null) {
                         this.payload.set(null);
                     } else {
-                        ByteBuf copyBuf = Unpooled.wrappedBuffer(data);
-                        final Object actualValue =
-                                Serializers.CORFU.deserialize(copyBuf, runtime);
-                        if (actualValue instanceof LogEntry) {
-                            ((LogEntry) actualValue).setGlobalAddress(getGlobalAddress());
-                            ((LogEntry) actualValue).setRuntime(runtime);
-                        }
-                        value = actualValue == null ? this.payload : actualValue;
-                        this.payload.set(value);
-                        copyBuf.release();
-                        lastKnownSize = data.length;
-                        data = null;
+                            ByteBuf copyBuf = Unpooled.wrappedBuffer(data);
+                            final Object actualValue;
+                            try {
+                                actualValue =
+                                        Serializers.CORFU.deserialize(copyBuf, runtime);
+                            } catch (Throwable throwable) {
+                                log.error("Exception caught at address {}, {}, {}",
```

Comment:
Need to release the buffer in the catch block. `copyBuf.release();`

Reply:
good catch. thanks!